### PR TITLE
Fixes #11865 partially

### LINKED ISSUE #11865: Support for shortcuts
The discussion here can be continued from #11862.

### PR DESCRIPTION
**Pull request by Achilles-96**
Signed-off-by: Raghuram Vadapalli raghuram.vadapalli@research.iiit.ac.in
Implemented shortcuts
- k - Toggle console
- h - Go to home page
- s - Open settings
- d + s - Go to database structure (Provided you are in database related page)
- d + f - Search database (Provided you are in database related page)
- t + s - Go  to table structure (Provided you are in table related page)
- t + f - Search table (Provided you are in table related page)

Update: backspace - Takes you to older page.

### CODE REVIEW

**Comment by zaverichintan** [comment]
Kindly use camel case

**Comment by nijel** [comment]
I think console activation should be also moved from it's scrip to this one...

**Comment by Achilles-96** [comment]
@nijel Do you mean toggle function?

**Comment by nijel** [comment]
I mean just the keyboard handling for that: https://github.com/phpmyadmin/phpmyadmin/blob/master/js/console.js#L130

**Comment by Achilles-96** [comment]
Any update on this?

**Comment by nijel** [comment]
I think this looks good. The only missing piece now is the documentation. I'd probably put that to the user guide section...

**Comment by Achilles-96** [comment]
Sure.

**Comment by Achilles-96** [comment]
By the way, I forgot to mention that backspace is for going back to earlier page.

File: libraries/config.default.php [hunk 1]
```diff
@@ -3133,3 +3133,10 @@
     'internal' => 50500,
     'human' => '5.5.0'
 );
+
+/**
+ * Disable shortcuts
+ *
+ * @global array $cfg['DisableShortcuts']
+ */
+$cfg['DisableShortcuts'] = false;
```
Comment (ibennetch) [inline libraries/config.default.php]: I think this might be better as DisableShortcutKeys, as it's a bit more readable.

**Comment by ibennetch** [comment]
I'm not sure what was intended here, but to get to a combination page (for example, database structure), I must press both keys at the same time (`d`+`s`). What I was expecting was to press first `d`, let go, then press `s`. Which one is better is debatable, but my preference is the individual sequence.

By the way, I like this feature more than I thought I would and find it quite handy already.

File: js/shortcuts_handler.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ * @name            Shortcuts handler
+ *
+ * @requires    jQuery
+ * @requires    jQueryUI
+ */
+
+/**
+ * This function is for parsing URL and getting required parameter
+ */
+function getUrlParameter(sParam) {
```
Comment (zixtor) [inline js/shortcuts_handler.js]: @Achilles-96 It seems that getUrlParameter is used just to get current database and table name values? then you could instead use PMA_commonParams.get('db') and PMA_commonParams.get('table').

**Comment by zixtor** [comment]
Though I would be okay with either way of key pressing, subsequent or simultaneous; but I would personally prefer simultaneous as that would be similar to usual Ctrl, Alt usage .

**Comment by zaverichintan** [comment]
I agree with @zixtor , as our users can easily adopt with ctrl alt system. As per my logic.

**Comment by Achilles-96** [comment]
Yes, what if they press d and they forget and press s for settings and they go to structure of database. I don't think that's a very good idea.

File: js/shortcuts_handler.js (hunk 2 above)
Comment (Achilles-96) [inline js/shortcuts_handler.js]: Sure, I didn't know you can do that.

**Comment by nijel** [comment]
There needs to be some (short) timeout. Try it on Github where I think it works within 2 seconds. If somebody forgets what he has pressed 2 seconds ago then it's different problem I think ;-).

**Comment by Achilles-96** [comment]
Fine. Implementing that is not a problem but are we sure we want to use subsequent instead of simultaneous.

**Comment by Achilles-96** [comment]
Please give an update.

**Comment by zixtor** [comment]
I think it will be great if both ways of using the key combination work, subsequent as well as simultaneous. For example in github also both ways seem to work.

**Comment by Achilles-96** [comment]
I agree. This is a better option.

**Comment by Achilles-96** [comment]
Is there any more objection to be addressed regarding this PR?

**Comment by nijel** [comment]
What is still missing is user documentation, otherwise I think it's good to merge.

**Comment by Achilles-96** [comment]
@nijel But aren't docs supposed to be in localized_docs repository

**Comment by nijel** [comment]
No, docs are in this repository, the other one (as it's name states) is for localized documentation only.

**Comment by Achilles-96** [comment]
Done.

**Comment by nijel** [comment]
Thanks! And my last request for you will be to improve commit message. It should describe the change in first, referencing fixed issue is nice to have, but this is something what will not help to identify reason for the change when browsing history...

**Comment by Achilles-96** [comment]
Is this commit message fine.

**Comment by nijel** [comment]
Thanks, it's better now.

**Pull request merged**